app/_src/gateway-operator/guides/autoscaling-workloads/datadog.md

```diff
@@ -0,0 +1,244 @@
+or use the following `values.yaml`:
+
+```yaml
+datadog:
+  kubelet:
+    tlsVerify: false
+
+clusterAgent:
+  enabled: true
+  # Enable the metricsProvider to be able to scale based on metrics in Datadog
+  metricsProvider:
+    # Set this to true to enable Metrics Provider
+    enabled: true
+    # Enable usage of DatadogMetric CRD to autoscale on arbitrary Datadog queries
+    useDatadogMetrics: true
+
+  prometheusScrape:
+    enabled: true
+    serviceEndpoints: true
+
+agents:
+  containers:
+    agent:
+      env:
+      - name: DD_HOSTNAME
+        valueFrom:
+          fieldRef:
+            fieldPath: spec.nodeName
+```
+
+to install [Datadog's helm chart][ddchart]:
+
+```bash
+helm repo add datadog https://helm.datadoghq.com
+helm repo update
+helm install -n datadog datadog --set datadog.apiKey=${DD_APIKEY} --set datadog.AppKey=${DD_APPKEY} datadog/datadog
+```
+
+[ddk8sguide]: https://docs.datadoghq.com/containers/kubernetes/installation/?tab=helm
+[ddchart]: https://github.com/DataDog/helm-charts/tree/main/charts/datadog
+
+## Generate traffic against `echo` `Service`
+
+Assuming that you have access to the deployed `Gateway` address you can try enforcing the scaling by issuing requests like so:
+
+```bash
+while curl -k "http://$(kubectl get gateway kong -o custom-columns='name:.status.addresses[0].value' --no-headers -n default)/echo/shell?cmd=sleep%200.1" ; do sleep 1; done
+```
+
+This will cause the underlying deployment to sleep for 100ms on each request and thus increase the average response time to that value.
+
+Keep this running while we move on to next steps.
+
+## Annotate {{ site.kgo_product_name }} with Datadog checks config
+
+{:.note}
+> **Note:** {{ site.kgo_product_name }} uses [kube-rbac-proxy][kuberbacproxy_github]
+> to secure its endpoints behind an RBAC proxy. That's why we'd going to scrape `kube-rbac-proxy`
+> and not the `manager` container.
+
+
+For Datadog to scrape {{ site.kgo_product_name }}'s metrics we need to let it know how to do it.
+This can be done through the following annotation on {{ site.kgo_product_name }}'s Pod:
+
+```yaml
+ad.datadoghq.com/kube-rbac-proxy.checks: |
+  {
+    "openmetrics": {
+      "instances": [
+        {
+          "bearer_token_auth": true,
+          "bearer_token_path": "/var/run/secrets/kubernetes.io/serviceaccount/token",
+          "tls_verify": false,
+          "tls_ignore_warning": true,
+          "prometheus_url": "https://%%host%%:8443/metrics",
+          "namespace": "autoscaling",
+          "metrics": [
+              "kong_upstream_latency_ms_bucket",
+              "kong_upstream_latency_ms_sum",
+              "kong_upstream_latency_ms_count",
+            ],
+          "send_histograms_buckets": true,
+          "send_distribution_buckets": true
+        }
+      ]
+    }
+  }
+```
+
+After applying the above you should see `avg:autoscaling.kong_upstream_latency_ms{service:echo}` metrics in your Datadog Metrics explorer.
```

Comment:
This metric is Datadog's `Distribution`: https://docs.datadoghq.com/metrics/distributions/.

I'm not an expert on this and for some reason this doesn't 100% track the value of an average across a designated time duration.

I've started [a thread in Datadog's slack](https://datadoghq.slack.com/archives/C3ZGCF88Z/p1709842615367649) about using 
 something like `sum:upstream_latency_ms.sum{service:echo} / sum:upstream_latency_ms.count{service:echo}` but for some reason I can't make this work. This query would better (as I look at it) track the average.